### ISSUE TITLE
'CREATE TABLE IF NOT EXISTS' -v- CREATE TABLE, could save error in CoreData etc.,

### ISSUE THREAD

**Issue by Qlogic** [sqlitebrowser/sqlitebrowser#875]
### Details for the issue

Hey, is there a way to do 'CREATE TABLE IF NOT EXISTS' with DB B, not just  CREATE TABLE, this could save error in CoreData and other, == SQLite error code:1, 'table  already exists'
#### I'm opening this issue because:

- [ ] DB4S is crashing
- [ ] DB4S has a bug
- [x ] DB4S needs a feature
- [ ] DB4S has another problem

#### I'm using DB4S on:

- [ ] Windows: ( _version:_ ___ )
- [ ] Linux: ( _distro:_ ___ )
- [x ] Mac OS: ( Sierre_version:_10.12.1 ___ )
- [ ] Other: ___

#### I'm using DB4S version:

- [x ] 3.9.1
- [ ] 3.9.0
- [ ] Other: ___

#### I have also:

- [ ] Tried out the nightly version: https://github.com/sqlitebrowser/sqlitebrowser#windows
- [x ] Searched for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Sorry for the delay, somehow I'd just not noticed this issue.

With the CoreData error you mention, what's the situation where it happens?  (just trying to understand the perspective you're coming from better) :smile:

**Comment by MKleusberg** [comment]
Hmm, that's not difficult to add but I wonder why it would be needed. In the Execute SQL tab you can type in whatever SQL statement you'd like. So I assume you don't mean that part of the program. And in the Create Table window, why would you want the ```IF NOT EXISTS``` bit? The statement wouldn't fail anymore but nothing would happen and you wouldn't even get an error message. I just imagine working on a table definition and, by accident, typing in the wrong table name, clicking OK, and all the work is lost :wink: Or maybe I'm missing something :smiley: Can you explain a little more, @Qlogic?

**Comment by justinclift** [comment]
Well, if cut-n-pasteing the output of some sqldump type backup commands, they generate `CREATE TABLE IF NOT EXISTS` style commands.  So this would be useful for handling that would needing people to manually edit the SQL.

While some of our user base would have no problem doing this editing, others would.  So if it's easy + accurate to implement we might as well. 😄

**Comment by Qlogic** [comment]
Sorry Martin,  This was a misunderstanding with me and Apple’s CoreData.  CoreData —no matter what I tried, showed up an error that my  table did not exist, when it did and I could see it by using SB. 

I was trying to pre-population a store and put it in my app bundle.
Turns out CoreData was creating the store for me and it was empty— no data—no table.

Once I got my head on straight and read their technical bulletin for the hundredth time, I figure out all I have to do was open their store with SB and fill it in with my data make a copy of that store and put it in my bundle, done.

Once again sorry the confusion,

Larry HansonSmith
larry@losttech.com
